core/include/scipp/core/transform.h

```diff
@@ -500,10 +541,33 @@ template <bool dry_run> struct in_place {
     }
     if constexpr (dry_run)
       return;
-    // WARNING: Do not parallelize this loop in all cases! The output may have a
-    // dimension with stride zero so parallelization must be done with care.
-    for (; std::get<0>(indices) != end; iter::increment(indices))
-      call_in_place(op, indices, arg, other...);
+    auto run = [&](auto indices, const auto &end) {
+      for (; std::get<0>(indices) != end; iter::increment(indices))
```

Comment:
won't this end up in an infinite loop if iterator == end? Similar issue below too.

Reply:
No? It first compares, then increments?

Reply:
What i mean is a scenario like
```
    auto it = a.end();
    it++;
    for(; it != a.end(); ++it){
    }
```
Which can happen if code external to this block is allowed to advance the iterator. This doesn't look like it can happen in the existing code so leave as is.

Reply:
Note that not all iterators support `<`, so it is quite common to use `!=` for iterators.